transportable-udfs-test/transportable-udfs-test-trino/src/main/java/com/linkedin/transport/test/trino/TrinoTester.java

```diff
@@ -7,46 +7,88 @@
 import com.linkedin.transport.test.spi.ToPlatformTestOutputConverter;
+import io.trino.spi.function.FunctionProvider;
+import io.trino.spi.type.Type;
+import io.trino.sql.SqlPath;
+import io.trino.sql.query.QueryAssertions;
+import io.trino.testing.LocalQueryRunner;
+import io.trino.testing.TestingSession;
+import io.trino.type.InternalTypeManager;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
 import static io.trino.type.UnknownType.UNKNOWN;
+import static org.assertj.core.api.Assertions.*;
 
 
-public class TrinoTester extends AbstractTestFunctions implements SqlStdTester {
+public class TrinoTester implements SqlStdTester {
 
   private StdFactory _stdFactory;
   private SqlFunctionCallGenerator _sqlFunctionCallGenerator;
   private ToPlatformTestOutputConverter _toPlatformTestOutputConverter;
+  private Session _session;
+  private FeaturesConfig  _featuresConfig;
+  private LocalQueryRunner _runner;
+  private QueryAssertions _queryAssertions;
 
   public TrinoTester() {
     _stdFactory = null;
     _sqlFunctionCallGenerator = new TrinoSqlFunctionCallGenerator();
     _toPlatformTestOutputConverter = new ToTrinoTestOutputConverter();
+    SqlPath sqlPath = new SqlPath("LINKEDIN.TRANSPORT");
+    _session = TestingSession.testSessionBuilder().setCatalog("LINKEDIN").setSchema("TRANSPORT").setPath(sqlPath).setClientCapabilities((Set) Arrays.stream(
+        ClientCapabilities.values()).map(Enum::toString).collect(ImmutableSet.toImmutableSet())).build();
+    _featuresConfig = new FeaturesConfig();
+    _runner = LocalQueryRunner.builder(_session).withFeaturesConfig(_featuresConfig).build();
+    _queryAssertions = new QueryAssertions(_runner);
   }
 
   @Override
   public void setup(
       Map<Class<? extends TopLevelStdUDF>, List<Class<? extends StdUDF>>> topLevelStdUDFClassesAndImplementations) {
+    Map<FunctionId, StdUdfWrapper> functions = new HashMap<>();
     // Refresh Trino state during every setup call
-    initTestFunctions();
     for (List<Class<? extends StdUDF>> stdUDFImplementations : topLevelStdUDFClassesAndImplementations.values()) {
       for (Class<? extends StdUDF> stdUDF : stdUDFImplementations) {
-        registerScalarFunction(new TrinoTestStdUDFWrapper(stdUDF));
+        StdUdfWrapper function = new TrinoTestStdUDFWrapper(stdUDF);
+        functions.put(function.getFunctionMetadata().getFunctionId(), function);
       }
     }
+    FunctionProvider functionProvider = new TransportFunctionProvider(functions);
+    ConnectorMetadata connectorMetadata = new TransportConnectorMetadata(functions);
+    Connector connector = new TransportConnector(connectorMetadata, functionProvider);
+    ConnectorFactory connectorFactory = new TransportConnectorFactory(connector);
+    _runner.createCatalog("LINKEDIN", connectorFactory, Collections.emptyMap());
```

Comment:
`_runner.createCatalog("LINKEDIN", connectorFactory, ImmutableMap.of("transport-udf-repo", "<TEST_UDF_REPO))` if we need to use another folder.

Reply:
As shown in the code above, TransportConnector used in this test is constructed independently. It is not created through TransportPlugin and TransportConnectorFactory where it loads UDF classes from the jars in the directory defined by "transport-udf-repo". Therefore we don't have to pass the value of "transport-udf-repo" here.